### PR TITLE
Update VOF mass conservation output

### PR DESCRIPTION
**Pull request by hepap**
# Description of the problem

- The mass conservation output parameters in VOF was tangled with the adaptive sharpening parameters.

# Description of the solution

- Clean the implementation so that both are independent.

# How Has This Been Tested?

Tests were updated as the output and parameters changed a bit, but the values remain the same

- [x] applications_tests/lethe-fluid/cavity_ht_average_velocity_restart.prm
- [x] applications_tests/lethe-fluid/gls_droplet_marangoni_effect.prm
- [x] applications_tests/lethe-fluid/gls_vof_1_isothermal_compressible_fluid.prm
- [x] applications_tests/lethe-fluid/gls_vof_2_isothermal_compressible_fluids.prm
- [x] applications_tests/lethe-fluid/gls_vof_dirichlet_boundary_condition.output
- [x] applications_tests/lethe-fluid/gls_vof_ht_heated-walls_heat-flux.prm
- [x] applications_tests/lethe-fluid/gls_vof_periodic_boundary_condition.prm
- [x] applications_tests/lethe-fluid/heat_transfer_vof_interface_gaussian_laser.prm
- [x] applications_tests/lethe-fluid/interface_sharpening_blurred_circle_adaptive_mass_conservation.prm
- [x] applications_tests/lethe-fluid/liquid_thermocapillary_droplet.prm
- [x] applications_tests/lethe-fluid/solid_thermocapillary_droplet.prm

 Restart was tested also.

# Documentation

Doc was updated.

- [x] vof.rst
- [x] postprocess.rst
- [x] rayleigh-taylor-instability.rst

### CODE REVIEW

**Review by AmishgaAlphonius — COMMENTED** [review]
Thanks for untangling the spaghetti!
Great job! I only have a few minor comments/suggestions :)

File: applications_tests/lethe-fluid/solid_thermocapillary_droplet.prm [hunk 1]
```diff
@@ -130,48 +130,48 @@ end
...
 #---------------------------------------------------
 
 subsection FEM
-  set velocity order = 1
-  set pressure order = 1
-  set VOF order      = 1
+  set velocity order    = 1
+  set pressure order    = 1
+  set VOF order         = 1
   set temperature order = 1
```
Comment (blaisb) [inline applications_tests/lethe-fluid/solid_thermocapillary_droplet.prm:175]: Do you need to explicitely add those?  I don't mind but it's really just a question.

File: examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-constant-sharpening.prm [hunk 2]
```diff
@@ -200,3 +204,15 @@ subsection linear solver
...
+  # Checkpointing parameters
+  set checkpoint = true
+  set frequency  = 10
+
+  # Output/input files
+  set filename   = restart
+
+  # Restarting parameter
+  set restart    = true
+end
```
[the remark this answers is not in the document]
Comment (blaisb) [reply examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-constant-sharpening.prm:218]: Wait is this test a restart or not a restart? If it's a restart then this section should be kept.

File: source/solvers/vof.cc [hunk 3]
```diff
@@ -628,10 +628,15 @@ VolumeOfFluid<dim>::finish_simulation()
...
   if (this_mpi_process == 0 &&
-      this->simulation_parameters.multiphysics.vof_parameters.conservation
-          .verbosity == Parameters::Verbosity::extra_verbose)
+      this->simulation_parameters.post_processing.verbosity ==
+        Parameters::Verbosity::verbose &&
+      this->simulation_parameters.post_processing.calculate_mass_conservation)
     {
+      std::cout << "+------------------------------------------+" << std::endl;
+      std::cout << "|           VOF Mass Conservation          |" << std::endl;
+      std::cout << "+------------------------------------------+" << std::endl;
```
Comment (blaisb) [inline source/solvers/vof.cc:639]: There is a function to make this kind of header.
https://github.com/lethe-cfd/lethe/blob/master/include/core/utilities.h#L318

You could use that if you want, it's a bit more "regular"
If you want to amke one that makes table like this, we could also use taht.

**Review by blaisb — APPROVED** [review]
2-3 tiny comments and questions. very good work @hepap

File: examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-constant-sharpening.prm (hunk 2 above)
[the remark this answers is not in the document]
Comment (AmishgaAlphonius) [reply examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-constant-sharpening.prm:218]: It is not a test, but an example, I think @hepap was testing the restart feature with the example when she finished and forgot to remove the subsection.

[the remark this answers is not in the document]
Comment (hepap) [reply examples/multiphysics/rayleigh-taylor-instability/rayleigh-taylor-instability-constant-sharpening.prm:218]: Amishga is rigth! I'll remove it :)

File: source/core/parameters_multiphysics.cc [hunk 4]
```diff
@@ -331,6 +265,20 @@ Parameters::VOF_InterfaceSharpening::parse_parameters(ParameterHandler &prm)
...
+    const std::string op_mf = prm.get("monitored fluid");
+    if (op_mf == "fluid 1")
+      monitored_fluid = Parameters::FluidIndicator::fluid1;
+    else if (op_mf == "fluid 0")
+      monitored_fluid = Parameters::FluidIndicator::fluid0;
+    else if (op_mf == "both")
+      monitored_fluid = Parameters::FluidIndicator::both;
+    else
+      throw(std::runtime_error("Invalid monitored fluid. "
+                               "Options are 'fluid 0' or 'fluid 1'."));
```
[the remark this answers is not in the document]
Comment (hepap) [reply source/core/parameters_multiphysics.cc:281]: In fact 'both' does not work! I removed it from the if condition :)

**Review by AmishgaAlphonius — APPROVED** [review]

**Pull request merged**